src/ekka_mnesia.erl

```diff
@@ -311,3 +318,19 @@ wait_for(tables) ->
         {error, Reason}      -> {error, Reason};
         {timeout, BadTables} -> {error, {timeout, BadTables}}
     end.
+
+%%--------------------------------------------------------------------
+%% Transaction API
+%%--------------------------------------------------------------------
+
+-spec(transaction(fun(() -> A)) -> mnesia:t_result(A)).
+transaction(Fun) ->
+    %% TODO: it should be possible initiate transactions on replicants too
+    core = ekka_rlog:role(),
+    %% TODO: this is wrong, we need to unwrap the result of the nested
```

Comment:
maybe we should not allow nested transactions at all?
i.e. add a level assert when inspecting mnesia activity store